webui/react/src/pages/ExperimentDetails/ExperimentTrials.tsx

```diff
@@ -305,6 +335,50 @@ const ExperimentTrials: React.FC<Props> = ({ experiment }: Props) => {
     }
   }, [ experiment.id, canceler, pagination, sorter, filters ]);
 
+  const sendBatchActions = useCallback((action: Action): Promise<void[] | CommandTask> => {
+    if (action === Action.OpenTensorBoard) {
+      return openOrCreateTensorboard(
+        { trialIds: selectedTrials.map(trial => trial.id) },
+      );
+    }
+    return Promise.all(selectedTrials.map(() => Promise.resolve()));
+  }, [ selectedTrials ]);
+
+  const clearSelected = useCallback(() => {
+    setSelectedRowKeys([]);
+  }, []);
+
+  const handleBatchAction = useCallback(async (action: Action) => {
+    try {
+      const result = await sendBatchActions(action);
+      if (action === Action.OpenTensorBoard && result) {
+        openCommand(result as CommandTask);
+      }
+
+      /*
+       * Deselect selected rows since their states may have changed where they
+       * are no longer part of the filter criteria.
+       */
+      clearSelected();
+
+      // Refetch experiment list to get updates based on batch action.
+      await fetchExperimentTrials();
```

Comment:
**suggestion:** maybe I'm not thinking about this right, but want to run this by you.

What if we don't clear the selection and skip fetching the trials upon opening up TensorBoards and keep preserving the previously selected Trials. Instead, we clear out the row selections when the user changes any of the filters, which I think is reasonable considering, when filtered the selection may no longer apply. I'm thinking for the use case of selecting some trials, view TB, and then thinking "oh want change the trials a bit because the previous selection was too noisy so I just want to change my previous selection".